internal/stores/contracts.go

```diff
@@ -287,3 +327,22 @@ func (s *SQLStore) contracts() ([]dbContract, error) {
 		Find(&contracts).Error
 	return contracts, err
 }
+
+func (s *SQLStore) AncestorContracts(id types.FileContractID, startHeight uint64) ([]bus.ArchivedContract, error) {
+	fcidGob, err := gobFCID(id)
+	if err != nil {
+		return nil, err
+	}
+	var ancestors []dbArchivedContract
+	err = s.db.Raw("WITH ancestors AS (SELECT * FROM archived_contracts WHERE renewed_to = ? UNION ALL SELECT archived_contracts.* FROM ancestors, archived_contracts WHERE archived_contracts.renewed_to = ancestors.fcid) SELECT * FROM ancestors WHERE start_height >= ?", fcidGob, startHeight).
```

Comment:
this would probably benefit from some indices, maybe `renewed_to` and `start_height` if that's possible?